### ISSUE TITLE
Creating a managed postgres db with custom password

### ISSUE THREAD

**Issue by janorn** [ansible/tower-operator#1052]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

When creating a postgres secret with only a custom password the ansible database_configuration tasks fails. According to the [readme ](https://github.com/ansible/awx-operator#external-postgresql-service) this should be possible.
```
resource "kubernetes_secret" "main" {
  metadata {
    namespace = var.kubernetes_namespace
    name      = "awx-postgres-configuration"
  }
  data = {
    "password" = random_password.password.result
    "type"     = "managed"
  }
}
```
```
      spec:
        postgres_configuration_secret: awx-postgres-configuration
```

https://github.com/ansible/awx-operator/blob/e8096a5f33dae40d0fbcc2cd6ad789e51dacb2d9/roles/installer/tasks/database_configuration.yml#L95-L103
```
TASK [installer : Store Database Configuration] ********************************
task path: /opt/ansible/roles/installer/tasks/database_configuration.yml:95
fatal: [localhost]: FAILED! => {\"censored\": \"the output has been hidden due to the fact that 'no_log: true' was specified for this result\"}

PLAY RECAP *********************************************************************
localhost                  : ok=31   changed=0    unreachable=0    failed=1    skipped=18   rescued=0    ignored=0
```

### AWX Operator version

0.28.0

### AWX version

Never got that far...

### Kubernetes platform

kubernetes

### Kubernetes/Platform version

v1.23.5

### Modifications

no

### Steps to reproduce

Create a custom postgres password.
```
resource "kubernetes_secret" "main" {
  metadata {
    namespace = var.kubernetes_namespace
    name      = "awx-postgres-configuration"
  }
  data = {
    "password" = random_password.password.result
    "type"     = "managed"
  }
}
```
```
      spec:
        postgres_configuration_secret: awx-postgres-configuration
```

### Expected results

A postgres pod running using my defined password.

### Actual results

Ansible fails configuring the database.

### Additional information

```
TASK [installer : Store Database Configuration] ********************************
task path: /opt/ansible/roles/installer/tasks/database_configuration.yml:95
fatal: [localhost]: FAILED! => {\"censored\": \"the output has been hidden due to the fact that 'no_log: true' was specified for this result\"}

PLAY RECAP *********************************************************************
localhost                  : ok=31   changed=0    unreachable=0    failed=1    skipped=18   rescued=0    ignored=0
```

### Operator Logs

```
TASK [installer : Store Database Configuration] ********************************
task path: /opt/ansible/roles/installer/tasks/database_configuration.yml:95
fatal: [localhost]: FAILED! => {\"censored\": \"the output has been hidden due to the fact that 'no_log: true' was specified for this result\"}

PLAY RECAP *********************************************************************
localhost                  : ok=31   changed=0    unreachable=0    failed=1    skipped=18   rescued=0    ignored=0
```

**Comment by janorn** [comment]
If i setup an external database and configure all fields it works fine.
Perhaps I missunderstand the managed type? If so I suggest to update the readme to clarify this.

> It is possible to set a specific username, password, port, or database, but still have the database managed by the operator. In this case, when creating the postgres-configuration secret, the type: managed field should be added.

**Comment by janorn** [comment]
Being able to configure the credentials would be useful.

**Comment by djyasin** [comment]
Hello,
We would like to gather a bit more information. If you could please set `no_log: 'false'` and run this again. Please let us know what failures you get with this revised setting.

**Comment by fosterseth** [comment]
@janorn were you able to resolve this issue?

**Comment by janorn** [comment]
No. This does not work. As can be seen by the ansible code cited. It will fail unless you populate all keys in the secret.
There are no defaults in the code.

**Comment by skattaa** [comment]
the secret name : postgres_configuration_secret is the problem .. the "_" is the issue it is against the naming scheme ! 

`The Secret "postgres_configuration_secret" is invalid: metadata.name: Invalid value: "postgres_configuration_secret": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
`
we should have the possibility to use another name to fix this issue
